docs/source/topics/processes/concepts.rst

```diff
@@ -23,16 +23,23 @@ A good thing to remember is that while it is running, we are dealing with the ``
 Process types
 =============
 
-Processes in AiiDA come in two flavors:
+In AiiDA, all processes are conceptually divided into two main types:
 
-* Calculation-like
-* Workflow-like
+* Calculation-like processes: These are processes that create data. Their role is to perform well-defined computations or transformations that produce new ``Data`` nodes, which are recorded as outputs (of type ``CREATE`` in the provenance graph).
+
+* Workflow-like processes: These are processes that orchestrate other processes, defining how multiple calculations or sub-workflows are executed in sequence or in parallel. Conceptually, workflows do not directly generate data, but rather return data produced by the calculations they run.
 
-The calculation-like processes have the capability to *create* data, whereas the workflow-like processes orchestrate other processes and have the ability to *return* data produced by calculations.
 Again, this is a distinction that plays a big role in AiiDA and is crucial to understand.
 For this reason, these different types of processes also get a different sub class of the ``ProcessNode`` class.
 The hierarchy of these node classes and the link types that are allowed between them and ``Data`` nodes, is explained in detail in the :ref:`provenance implementation<topics:provenance:implementation>` documentation.
 
+
+.. note:: Technically, a workflow is able to create new ``Data`` nodes directly (for example, by instantiating and storing a ``Dict`` or ``StructureData`` node inside its code). However, this will *not* show up as a node created by the workflow in the provenance graph, i.e., there will be no direct link between the workflow node and the ``Data`` node. Instead, the ``Data`` node will appear as having no creator (similar to a ``Data`` node created in the shell). 
```

Comment:
```suggestion
.. note:: Technically, a workflow is able to create new ``Data`` nodes directly (for example, by instantiating and storing a ``Dict`` or ``StructureData`` node inside its code). However, this will *not* show up as a node created by the workflow in the provenance graph, and the ``Data`` node will appear as having no creator (similar to a ``Data`` node created in the shell). 
```

You will/can get a "direct" RETURN link from the workflow to the data node, so perhaps removing this part is better.